src/datasources/cache/cache.first.data.source.ts

```diff
@@ -36,18 +36,20 @@ export class CacheFirstDataSource {
    * Gets the cached value behind the {@link CacheDir}.
    * If the value is not present, it tries to get the respective JSON
    * payload from {@link url}.
-   * 404 errors are cached with {@link ERROR_TTL_SECONDS} seconds expiration time.
+   * 404 errors are cached with {@link notFoundErrorTTLSeconds} seconds expiration time.
    *
    * @param cacheDir - {@link CacheDir} containing the key and field to be used to retrieve from cache
    * @param url - the HTTP endpoint to retrieve the JSON payload
    * @param networkRequest - the HTTP request to be used if there is a cache miss
    * @param expireTimeSeconds - the time to live in seconds for the payload behind {@link CacheDir}
+   * @param notFoundErrorTTLSeconds - the time to live in seconds for the error when the item is not found
    */
   async get<T>(
     cacheDir: CacheDir,
     url: string,
     networkRequest?: NetworkRequest,
     expireTimeSeconds?: number,
+    notFoundErrorTTLSeconds?: number,
```

Comment:
If we extract `defaultNotFoundErrorTTLSeconds` then we don't need to have this parameter as `undefined` which I think it's preferable.

Reply:
Yes, I also thought the same, but I defined this as optional to stay consistent with `expireTimeSeconds`, which is optional but it's never called with an `undefined` value in the code. Maybe we should make both non-optional? What do you think? @fmrsabino

Reply:
But `expireTimeSeconds` is not required for some endpoints right?

Reply:
Right, that should be optional, but that's not the case for `notFoundExpireTimeSeconds`, which can be mandatory and will be explicitly set by the caller (`TransactionApi`, `ConfigApi`, etc.) 👍

Reply:
I've made it mandatory in [4e30dc6](https://github.com/safe-global/safe-client-gateway-nest/pull/578/commits/4e30dc61d72b12d7b8baa789be526cca80bca083)